src/libraries/System.Private.CoreLib/src/System/IO/RandomAccess.Unix.cs

```diff
@@ -86,7 +86,10 @@ internal static unsafe int WriteAtOffset(SafeFileHandle handle, ReadOnlySpan<byt
                 // The Windows implementation uses WriteFile, which ignores the offset if the handle
                 // isn't seekable.  We do the same manually with PWrite vs Write, in order to enable
                 // the function to be used by FileStream for all the same situations.
-                int result = handle.CanSeek ?
+
+                // POSIX requires that pwrite should respect provided offset even for handles opened with O_APPEND.
+                // But Linux and BSD don't do that, moreover their behaviour is different. So we always use write for O_APPEND.
+                int result = handle.CanSeek && !handle.IsAppend ?
```

Comment:
Doesn't this add yet another in some cases (in particular for a `new SafeFileHandle(fd, ...)`-constructed handle?  Is that really worth it?